host/src/include_private/nkv_framework.h

```diff
@@ -282,16 +286,28 @@
         pthread_rwlock_init(&data_rw_lock_list[iter], NULL);
       }
 
-      listing_keys = new std::unordered_map<std::size_t, std::set<std::string> > (nkv_listing_cache_num_shards);
+      // Resize and initialize std::unordered_map at each index
+      listing_keys.resize(nkv_listing_cache_num_shards);
+      //listing_keys = new std::unordered_map<std::size_t, std::set<std::string> > (nkv_listing_cache_num_shards);
+      for (auto i =0; i<nkv_listing_cache_num_shards; i++) {
+          // Initialize map at index
+          listing_keys[i] = std::unordered_map<std::size_t, std::set<std::string>>();
+      }
       if (nkv_in_memory_exec) {
-        data_cache = new std::unordered_map<std::string, nkv_value_wrapper*> (nkv_listing_cache_num_shards);
+        //data_cache = new std::unordered_map<std::string, nkv_value_wrapper*> (nkv_listing_cache_num_shards);
+        // Resize and initialize data_cache
+        data_cache.resize(nkv_listing_cache_num_shards);
+        for (auto i=0; i<nkv_listing_cache_num_shards; i++) {
+            // Initialize map at index
+            data_cache[i] = std::unordered_map<std::string, nkv_value_wrapper*>();
```

Comment:
How nkv_value_wrapper* will be deleted ?

Reply:
Fixed